### PR TITLE
[VERSION_APITEST] Add tests for VerQueryValue(A/W)

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Add tests for VerQueryValue(A/W)

JIRA issue: [CORE-19783](https://jira.reactos.org/browse/CORE-19783)

## Tests

- ✅ Test WHS: https://reactos.org/testman/compare.php?ids=98499
- ✅ Test Windows 2003 x64: https://reactos.org/testman/compare.php?ids=98491

### CODE REVIEW

**Review by Doug-Lyons — APPROVED** [review]

**Review by whindsaks — APPROVED** [review]

**Review by learn-more — APPROVED** [review]

File: modules/rostests/apitests/version/VerQueryValue.c [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    PVOID pvData = NULL;
+    SIZE_T ExpectedOffset;
+    UINT cbLen;
+    BOOL result;
+
+    /* Make a copy of the version info. Windows actually writes to it!
+       We make the buffer twice as big to account for the ANSI copy,
+       even if we don't use it. */
+    pVersionInfo = malloc(2 * sizeof(g_VersionInfo));
+    memset(pVersionInfo, 0, 2 * sizeof(g_VersionInfo));
```
Comment (HBelusca) [inline modules/rostests/apitests/version/VerQueryValue.c:250]: or malloc + memset(..., 0, ...)  --> calloc ?

File: modules/rostests/apitests/version/VerQueryValue.c [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    USHORT uLanguage, uCodePage;
+    UINT cbLen;
+    BOOL result;
+
+    /* Get the size for the unicode version info */
+    dwVersionInfoSizeW = GetFileVersionInfoSizeW(L"kernel32.dll", NULL);
+    ok(dwVersionInfoSizeW > 0, "GetFileVersionInfoSizeW failed\n");
+
+    /* Get the unicode version info from kernel32 */
+    pvVersionInfoW = malloc(dwVersionInfoSizeW);
```
Comment (HBelusca) [inline modules/rostests/apitests/version/VerQueryValue.c:209]: The ansi test also add a filling of the allocated buffer with `0xCC`.  Why not doing it there as well?

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]

File: modules/rostests/apitests/version/VerQueryValue.c (hunk 1 above)
Comment (tkreuzer) [reply modules/rostests/apitests/version/VerQueryValue.c:250]: I don't think that is an improvement

File: modules/rostests/apitests/version/VerQueryValue.c (hunk 2 above)
Comment (tkreuzer) [reply modules/rostests/apitests/version/VerQueryValue.c:209]: The ansi test calls `GetFileVersionInfoA` to fill the buffer appropriately. The static test passes something that is already supposed to be filled in.

**Pull request merged**